ngraph/core/include/ngraph/op/if.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+
+#include "ngraph/function.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/util/multi_subgraph_base.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v8
+        {
+            /// \brief  If operation.
+            class NGRAPH_API If : public util::MultiSubGraphOp
+            {
+            public:
+                enum BodyIndexes
+                {
+                    then_body_index = 0,
+                    else_body_index = 1
+                };
+
+                NGRAPH_RTTI_DECLARATION;
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                /// \brief     Constructs If with condition
+                ///
+                /// \param     execution_condition   condition node.
+                If(const Output<Node>& execution_condition);
+                If();
+                explicit If(const OutputVector& values);
```

Comment:
why don't we have `then` and `else` body functions? I think they must be set immediately one you create `IF` function

Reply:
Good question. This was done similarly to Loop. This constructor  would only be needed only for unit_tests. Such constructor with multiple parameters would look ugly.

Reply:
it happened historically:)

similar to TensorIterator / Loop ops, we create If op step by step. 
usually:
ctor()
set_body()
add_input_desc()
add_output_desc()
...

Reply:
Is this constructor really needed?

Reply:
This constructor was deleted.